### ISSUE TITLE
tensorflow preventing container to run without AVX

### ISSUE THREAD

**Issue by baramuse** [IntelAI/OpenVINO-model-server#49]
When trying to run the container on a QNAP TS-251, having a Celeron J1800, the container would not start because of missing AVX instructions :

`[tensorflow/core/platform/cpu_feature_guard.cc:37] The TensorFlow library was compiled to use AVX instructions, but these aren't available on your machine.`                                                                                                                                                                                       

I've been running inference with OpenVino docker image on this NAS without any problem, why should TensorFlow prevent running the whole OpenVino server while it's only there for the protobuf API format right ?

**Comment by dtrawins** [comment]
In the OpenVINO model server is used several functions for tensorflow contrib which implement data serialization and deserialization in a proto buffer. The inference itself is used without tensorflow library.
In your case the easiest fix should be to build the pip package for tensorflow without AVX support. 
You could probably use the package described in https://github.com/yaroslavvb/tensorflow-community-wheels/issues/103 
I don't have any host without AVX at hand to test it but it should work if you replace the tensorflow dependency in the requirements section, with the pip package without avx.
Please confirm if that can work for you.

**Comment by baramuse** [comment]
Thanks for pointing the way
I created a public image on docker.io :  smartcv/openvino-model-server-noavx:0.0.1

Now when I pull and run this image, I get strictly no logs.

I create the container like that : 
`docker run --rm -d  -v /share/CACHEDEV1_DATA/Public/ml/:/ml:ro --env LOG_LEVEL=DEBUG --env LOG_PATH=/var/log/ie_serving.log -p 9001:9001 smartcv/openvino-model-server-noavx:0.0.1 \
/ie-serving-py/start_server.sh ie_serving model --model_path /ml/models/cockburn_ssdlite --model_name cockburn --port 9001`

I can only see that the process is running : 
`root         1  1.9  4.1 586752 164268 ?       Ss   10:00   0:04 /ie-serving-py/.venv/bin/python3 /ie-serving-py/.venv/bin/ie_serving model --model_path /ml/models/cockburn_ssdlite --model_name cockburn --port 9001`

nothing in /var/logs/ie_serving.log, nothing in the container's logs..

Any idea ?

**Comment by dtrawins** [comment]
This is strange indeed. I tested your docker image and it worked fine for me.
I suggest to start the container in interactive mode `docker run --rm -it` with just bash command and then check what happens when you start the service manually. There should be startup messages on the console and errors if some problems occurred.

**Comment by baramuse** [comment]
I must remind I'm trying to run it on a QNAP NAS ts-251, with container station. (Docker version 17.09.1-ce, build 0a5102e)

```
[~] # docker run --rm -it  b7919426c99c /bin/bash
root@32e8fdfb2920:/ie-serving-py#./start_server.sh ie_serving model --model_path /ml/models/cockburn_ssdlite --model_name cockburn --port 9001
```
nothing happens, no log anywhere...

**Comment by dtrawins** [comment]
What is the CPU type that you have on board?
The symptoms are quite strange. I suspect python environment is not starting.
Can you try interactively start python:
`./start_server.sh python` inside the docker and try to import the dependencies:
```import argparse
import json
import sys
from ie_serving.models.model_builder import ModelBuilder
from ie_serving.server.start import serve as start_server
from ie_serving.logger import get_logger, LOGGER_LVL
from jsonschema.exceptions import ValidationError
import os```
That might reveal which dependency hangs on the import attempt.
BTW Do other docker containers work correctly on this device? Are you able to reproduce it on different device?

**Comment by baramuse** [comment]
The CPU on this NAS is a Celeron J1800.
I do have other containers running without issues.

I ran interactively the imports, it looks like it's stalling when import  ie_serving.server.start :
```
root@32e8fdfb2920:/ie-serving-py# source .venv/bin/activate                                                                                                                                                                                                  
(.venv) root@32e8fdfb2920:/ie-serving-py# python                                                                                                                                                                                                             
Python 3.5.2 (default, Nov 12 2018, 13:43:14)                                                                                                                                                                                                                
[GCC 5.4.0 20160609] on linux                                                                                                                                                                                                                                
Type "help", "copyright", "credits" or "license" for more information.                                                                                                                                                                                       
>>> import json                                                                                                                                                                                                                                              
>>> import sys                                                                                                                                                                                                                                               
>>> from ie_serving.models.model_builder import ModelBuilder                                                                                                                                                                                                 
>>> from ie_serving.server.start import serve as start_server 
```

Does not go any further.

I'll try the same image on another (same model) NAS and on a different CPU NAS (but still with no AVX which is the whole point here :) )

**Comment by baramuse** [comment]
So I went one step farther, I interactively ran the imports of ie_serving.server.start :

```
(.venv) root@96688c4aea33:/ie-serving-py# python
Python 3.5.2 (default, Nov 12 2018, 13:43:14)
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from concurrent import futures
>>> import time
>>> import grpc
>>> import tensorflow.contrib.util as tf_contrib_util
```

So looks like the culprit here is the tensorflow package.. which makes sens as it's the only thing changes in this build of the docker.

I'll try to find a build of tensorflow that runs on this CPU, then will re-build a docker image with it.

Thanks for pointing out directions to troubleshoot !

**Comment by baramuse** [comment]
Got it working with another Tensorflow build from this request : https://github.com/yaroslavvb/tensorflow-community-wheels/issues/98

I published the version 0.0.2 with working tensorflow.

Thanks again for your help !
